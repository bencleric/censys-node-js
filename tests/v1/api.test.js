const { setupRecorder } = require("nock-record");
const { ClientV1 } = require("../../src");
const { API_ID, API_SECRET } = require("../consts");
const { afterRecord } = require("../utils");

const record = setupRecorder();

describe("#censys.v1.client", () => {
  const c = new ClientV1(API_ID, API_SECRET);

  it("get account data", async () => {
    const { completeRecording } = await record("censys-v1-account", {
      afterRecord,
    });
    const res = await c.ipv4.account();
    completeRecording();
    expect(res).toMatchSnapshot();
  });
});
