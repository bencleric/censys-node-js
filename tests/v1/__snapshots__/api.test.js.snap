// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#censys.v1.client get account data 1`] = `
Object {
  "email": "test@censys.io",
  "first_login": "2020-01-12 12:00:00",
  "last_login": "2021-01-12 12:00:00.000000",
  "login": "test@censys.io",
  "quota": Object {
    "allowance": 500000,
    "resets_at": "2021-01-12 12:00:00",
    "used": 0,
  },
}
`;
